standards/coding-standards.md

```diff
@@ -1,9 +1,31 @@
 # Coding Standards
 
-Apps MAY enforce a rigorous coding style.  
-Different languages have different opinions about uniformity.
+The Digital department at NYPL was formed from several different teams, that had generally been independent and did not share standards on technology, etc. This means that the systems that Digital is responsible for use many different languages, frameworks, deployment environments etc., etc.
 
-If a project wants to enforce a style - it SHOULD be called
+This can naturally lead to some issues, as not everyone can be conversant in every technology that is used (while this problem is certainly not unique to NYPL it is particularly pronounced here). This in turn can lead to resource constraints in planning, as you don't just need to worry about what team should deal with something, you also need to consider the individual skillsets. Additionally it can also cause problems when creating new services — should we use Rails, node, Python, Java, PHP? Microservice or monolith? 
+
+Therefore, to help inform any technology decisions in the future we have the following guiding principles:
+
+* Except in particular, specialist areas (mobile development, SimplyE) new development should use JavaScript or Python.
+* If you wish to use a different language, you should discuss with Engineering Managers first, and have considered how the thing you are developing will be maintained in future.
+* We like microservice architectures, which should use RESTful APIs and/or event streams or message queues (the wide variety of languages and frameworks we use make microservices more attractive, as we are less constrained by prior use of languages than we would be in a monolithic application, and, at worst, it is easier to rewrite a microservice in a different language than a larger app).
+* We prefer our RESTful APIs to use JSON as the main data exchange format
+* All APIs we create should have Swagger Docs endpoints
```

Comment:
Should we say Swagger 2.0 or OpenAPI 3.0? (The actual swagger standard has been pretty old for a while now and we might want to consider migrating to the newer version. Not sure what compatibility issues that poses.)